core/src/autogluon/core/hpo/executors.py

```diff
@@ -181,13 +181,17 @@ def register_resources(
                 }
         if 'resources_per_trial' not in self.hyperparameter_tune_kwargs:
             # User didn't provide any requirements
-            model_estimate_memory_usage = None
             num_jobs_in_parallel_with_mem = math.inf
 
             if initialized_model.estimate_memory_usage is not None:
-                model_estimate_memory_usage = initialized_model.estimate_memory_usage(**kwargs)
-                total_memory_available = psutil.virtual_memory().available
-                num_jobs_in_parallel_with_mem = total_memory_available // model_estimate_memory_usage
+                @disable_if_lite_mode(ret=num_jobs_in_parallel_with_mem)
+                def _update_num_jobs_in_parallel_with_mem():
+                    import psutil
+                    model_estimate_memory_usage = initialized_model.estimate_memory_usage(**kwargs)
+                    total_memory_available = psutil.virtual_memory().available
```

Comment:
Just out of curiosity, why not just use available_virtual_mem() directly, instead of using disable_if_lite_mode decorator for the whole function?

Reply:
nice catch.

Reply:
@yinweisu is it a problem to have wrappers like this for distributed training? I remember some decorators around serializable objects was causing problems when I started do distributed folds work.

Reply:
I just tried the above code with ray and it works fine:
```python
import ray
import psutil

def disable_if_lite_mode(ret=None):
    def inner(func):
        def do_nothing(*args, **kwargs):
            if callable(ret):
                return ret(*args, **kwargs)
            return ret
        return do_nothing
    return inner

@disable_if_lite_mode(ret=1)
def dummy():
    return psutil.cpu_count()

@ray.remote
def test():
    return dummy()
    
ray.init()
futures = [test.remote() for _ in range(4)]
print(ray.get(futures))
```
OUTPUTS:
```bash
(cloud) ubuntu@ip-172-31-11-12:~/yinweisu/test_scripts$ python3 temp.py 
2022-11-21 20:27:51,964 INFO worker.py:1518 -- Started a local Ray instance.
[1, 1, 1, 1]
```